modules/azure/storage_account_public/variables.tf

```
@@ -99,4 +99,13 @@ variable "static_website" {
   })
   description = "Describes the static website configuration for the storage account"
   default     = null
+}
+
+variable "custom_domain" {
+  type = object({
+    name          = string,
+    use_subdomain = bool
```

Should this not be optional like described in the docs?
https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_account#use_subdomain

> use_subdomain - (Optional) Should the Custom Domain Name be validated by using indirect CNAME validation?

---

```suggestion
    use_subdomain = optional(bool)
```

I don't know if this works because the docs don't tell what the default value is, but this is the best thing i can think of.

---

I have tested this and it works